index.js

```diff
@@ -239,67 +205,68 @@ class Paper {
     /**
      * Renders a template with the given context
      *
-     * @param {String} path
-     * @param {Object} context
-     * @return {String}
+     * @param {String} path The path to the template
+     * @param {Object} context The context to provide to the template
+     * @return {Promise} A promise to return the rendered template
      * @throws [TemplateNotFoundError|RenderError|DecoratorError]
      */
     render(path, context) {
         return this.renderer.render(path, context);
     }
 
     /**
-     * Theme rendering logic
+     * Theme rendering logic. This is used by Stapler and Stencil CLI.
      *
-     * @param  {String|Array} templatePath
+     * @param  {String|Array} templatePath A single template or list of templates to render.
      * @param  {Object} data
-     * @return {String|Object}
+     * @return {Promise} A promise that returns a {String|Object} depending on whether multiple templates were rendered
      * @throws [TemplateNotFoundError|RenderError|DecoratorError]
      */
     renderTheme(templatePath, data) {
-        let html;
-        let output;
-
-        // Is an ajax request?
-        if (data.remote || _.isArray(templatePath)) {
+        // Simple case of a single non-ajax template
+        if (!data.remote && !_.isArray(templatePath)) {
+            return this.render(templatePath, data.context);
+        }
 
-            if (data.remote) {
-                data.context = Object.assign({}, data.context, data.remote_data);
-            }
+        // If no template path provided, just return the remote data
+        if (!templatePath) {
```

Comment:
Shouldn't this `if` check be placed before ` if (!data.remote && !_.isArray(templatePath))`

Reply:
No, I think we need to assume at this point that it is a remote request.